ml-agents-envs/mlagents_envs/base_env.py

```diff
@@ -22,7 +22,7 @@
 import numpy as np
 from enum import Enum
 
-AgentId = int
+AgentId = Union[int, str]
```

Comment:
At this point of the code, agentIds are always int.

Reply:
Would it be valid to redefine the `AgentId` type in env_manager? I also thought about finding a way to append a worker ID to an int

Reply:
I would define another one yes. Ideally not a string but a namedtuple pair (environmentId, agentId)

Reply:
What's ugly here is that we'd have to make a new type of BatchedStepResult since the field would have to mutate to a different type. So like EnvBatchedStepResult which contains a namedtuple or string as the `agent_id` field. Then everything from the env_manager onwards takes this type instead of BatchedStepResult. 

Perhaps this is OK.

Reply:
What if we just keep it as ints and concat the ints (e.g. agent_id 1234 worker_id 2 becomes agent_id 21234)?

Reply:
> e.g. agent_id 1234 worker_id 2 becomes agent_id 21234

So would worker_id 21 and agent_id 234

Reply:
agent IDs are 32 bits in unity. How about using `(worker_id << 32) | agent_id`?

Reply:
Oh interesting, and `int` in Python is 64. Guess this might work... but then all of our np.ndarrays are 32 bit :D

We decided to have a separate worker_id and only assemble the final agent_id in the AgentProcessor as a `str` as we have been before. Seems like it's the cleanest way to do it for now.